pkg/config/generate.go

```diff
@@ -32,15 +33,10 @@ func createUserHtPassEntry(name string, passwd *string, htPasswd []byte) ([]byte
 	return htPassEntry, nil
 }
 
-func (g *simpleGenerator) Generate(cs *api.OpenShiftManagedCluster) (err error) {
+func (g *simpleGenerator) Generate(cs *api.OpenShiftManagedCluster, template *pluginapi.Config) (err error) {
+	cs.Config = api.ConvertFromPlugin(*template)
```

Comment:
This line is problematic for upgrades, that's why the re-entrancy test fails to find the storage blob. You should be able to reuse the existing config and merge on top of it the template.

Reply:
At this point, cs.Config is already populated with the old config so really this function should not be a conversion but a merger.

Reply:
ack! thanks for saving me a hassle :)